### PR TITLE
feat: extend watermark API to get both low/high offset

### PR DESCRIPTION
**Pull request by crepererum**
Based on #99.

### CODE REVIEW

File: src/client/partition.rs [hunk 1]
```diff
@@ -585,16 +593,20 @@ fn extract_records(
     Ok(records)
 }
 
-fn build_list_offsets_request(partition: i32, topic: &str) -> ListOffsetsRequest {
+fn build_list_offsets_request(partition: i32, topic: &str, at: OffsetAt) -> ListOffsetsRequest {
+    let timestamp = match at {
+        OffsetAt::Earliest => -2,
```
Comment (tustvold) [inline src/client/partition.rs:598]: :magic_wand:

**Review by tustvold — APPROVED** [review]
I'm guessing this would only be useful for reporting, as the earliest offset may change before a subsequent fetch records call, leading to OffsetOutOfRange

**Comment by crepererum** [comment]
> I'm guessing this would only be useful for reporting, as the earliest offset may change before a subsequent fetch records call, leading to OffsetOutOfRange

This is true for both offsets, I'll add a note/warning to the method.

**Comment by crepererum** [comment]
> > I'm guessing this would only be useful for reporting, as the earliest offset may change before a subsequent fetch records call, leading to OffsetOutOfRange
> 
> This is true for both offsets, I'll add a note/warning to the method.

done

**Pull request merged**